datafusion/src/physical_plan/planner.rs

```diff
@@ -749,32 +754,9 @@ impl DefaultPhysicalPlanner {
                     "Unsupported logical plan: CreateExternalTable".to_string(),
                 ))
             }
-            LogicalPlan::Explain {
```

Comment:
previously the physical planner could handle an `LogicalPlan::Explain` anywhere in the tree. However, I can't think if cases where this can actually occur and it made adding the final physical plan challenging. 

So I have changed the code to special case handling `Explain` as the root.